src/urlencoded.rs

```diff
@@ -30,20 +30,20 @@ impl Params {
 }
 
 pub fn parse(encoded_string : &str) -> Params {
-    Params(form_urlencoded::parse(encoded_string.as_bytes()).into_iter().group())
+    Params(form_urlencoded::parse(encoded_string.as_bytes()).into_owned().group())
 }
 
 pub fn parse_uri(origin: &RequestUri) -> Params {
-    let f = |query: Option<&String>| query.map(|q| parse(&*q));
+    let f = |query: Option<&str>| query.map(|q| parse(&*q));
 
     let result = match *origin {
-        AbsoluteUri(ref url) => f(url.query.as_ref()),
-        AbsolutePath(ref s) => UrlParser::new().parse_path(&*s)
-                                                // FIXME: If this fails to parse,
-                                                // then it really shouldn't have
-                                                // reached here.
-                                               .ok()
-                                               .and_then(|(_, query, _)| f(query.as_ref())),
+        AbsoluteUri(ref url) => f(url.query()),
+        AbsolutePath(ref s) => {
+            let base = "f://".parse().unwrap();
```

Comment:
The protocol here doesn't actually matter, we just need some arbitrary base URL to parse the absolute path off of since rust-url dropped support for parsing subcomponents of the URL (is that right, @SimonSapin?).